src/Features/Core/Portable/Completion/Providers/SymbolCompletionItem.cs

```diff
@@ -59,11 +59,14 @@ public static CompletionItem AddSymbolEncoding(IReadOnlyList<ISymbol> symbols, C
             return item.AddProperty("Symbols", EncodeSymbols(symbols));
         }
 
-        public static CompletionItem AddSymbolNameAndKind(IReadOnlyList<ISymbol> symbols, CompletionItem item)
+        public static CompletionItem AddSymbolInfo(IReadOnlyList<ISymbol> symbols, CompletionItem item)
         {
             var symbol = symbols[0];
+            var isGeneric = symbol.GetArity() > 0;
+
             return item.AddProperty("SymbolKind", ((int)symbol.Kind).ToString())
-                       .AddProperty("SymbolName", symbol.Name);
+                       .AddProperty("SymbolName", symbol.Name)
+                       .AddProperty("IsGeneric", isGeneric.ToString());
```

Comment:
Can we just not add `IsGeneric` when it's not a generic type, and use the existence of the property as the indication instead? This way we would avoid creating a new item for majority of the items that are not generic.

Reply:
Sure. Fixed.